File: packages/drivekit_core/flutter_drivekit_core_ios/ios/Classes/DrivekitCorePlugin.swift
import Flutter
import UIKit
import DriveKitCoreModule

extension FlutterError: Error {}

public class DrivekitCorePlugin: NSObject, FlutterPlugin, IOSCoreApi {
    public static func register(with registrar: FlutterPluginRegistrar) {
        let messenger: FlutterBinaryMessenger = registrar.messenger()
        let api: IOSCoreApi & NSObjectProtocol = DrivekitCorePlugin.init()

        IOSCoreApiSetup.setUp(binaryMessenger: messenger, api: api)
    }

    public func getPlatformName() throws -> String {
        return "iOS"
    }

    public func getApiKey() throws -> String? {
        return DriveKit.shared.config.getApiKey()
    }

    public func setApiKey(key: String) {
        DriveKit.shared.setApiKey(key: key)
    }

    public func setUserId(userId: String) {
        DriveKit.shared.setUserId(userId: userId)
    }

    public func getUserId() -> String? {
        return DriveKit.shared.config.getUserId()
    }

    public func reset() {
        DriveKit.shared.reset()
    }

    func isTokenValid() throws -> Bool {
        return DriveKit.shared.isTokenValid()
    }

    func deleteAccount(instantDeletion: Bool) throws {
        DriveKit.shared.deleteAccount(instantDeletion: instantDeletion)
    }

    func disableLogging(showInConsole: Bool) throws {
        DriveKit.shared.disableLogging(showInConsole: showInConsole)
    }
}

File: packages/drivekit_core/flutter_drivekit_core_android/android/src/main/kotlin/com/drivekit/core/DrivekitCorePlugin.kt
package com.drivekit.core

import AndroidCoreApi
import android.content.Context
import io.flutter.embedding.engine.plugins.FlutterPlugin
import com.drivequant.drivekit.core.DriveKit

class DrivekitCorePlugin : FlutterPlugin, AndroidCoreApi {
    private var context: Context? = null


    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        AndroidCoreApi.setUp(flutterPluginBinding.binaryMessenger, this)
        context = flutterPluginBinding.applicationContext
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        AndroidCoreApi.setUp(binding.binaryMessenger, null)
        context = null
    }

    override fun getPlatformName(): String {
        return "android"
    }

    override fun setApiKey(key:String) {
       DriveKit.setApiKey(key)
    }

    override fun setUserId(userId:String) {
        DriveKit.setUserId(userId)
    }

}
